pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -115,26 +161,36 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 	if service.Annotations == nil {
 		service.Annotations = map[string]string{}
 	}
-	if infraConfig.Status.Platform == configv1.AWSPlatformType {
-		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
-		// Set the load balancer for AWS to be as aggressive as Azure (2 fail @ 5s interval, 2 healthy)
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval"] = "5"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout"] = "4"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold"] = "2"
-		service.Annotations["service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold"] = "2"
-	}
-	// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
-	// GCP load balancers are not customizable and are set to (3 fail @ 8s interval, 1 healthy)
 
-	if infraConfig.Status.Platform == configv1.IBMCloudPlatformType {
-		service.Annotations[iksLBScopeAnnotation] = "public"
+	if proxyNeeded {
+		service.Annotations[awsLBProxyProtocolAnnotation] = "*"
 	}
 
-	if isInternal {
-		annotation := InternalLBAnnotations[infraConfig.Status.Platform]
-		for name, value := range annotation {
-			service.Annotations[name] = value
+	if platform != nil {
+		if isInternal {
+			annotation := InternalLBAnnotations[platform.Type]
+			for name, value := range annotation {
+				service.Annotations[name] = value
+			}
+		}
+		switch platform.Type {
+		case configv1.AWSPlatformType:
+			if ci.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.Type == operatorv1.AWSLoadBalancerProvider &&
+				ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type == operatorv1.AWSNetworkLoadBalancer {
```

Comment:
Need a nil check for `ci.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS`.